test/model/src/driver/kv_workload.cpp

```diff
@@ -37,6 +37,168 @@ extern "C" {
 
+namespace operation {
+
+/*
+ * parse --
+ *     Parse an operation from a string. Throw an exception on error.
+ */
+any
+parse(const char *str)
+{
+    const char *p = str;
+
+    /* Get the operation name. */
+    std::string name;
+    while (*p != '\0' && (std::isalnum(*p) || *p == '_'))
+        name += *(p++);
+
+    while (*p != '\0' && std::isspace(*p))
+        p++;
+    if (*(p++) != '(')
+        throw model_exception("Expected '('");
+
+    /* Get the arguments. */
+    std::vector<std::string> args;
+    std::string current;
+    bool done = false;
+    bool escaped = false;
+    bool had_quotes = false;
+    bool quotes = false;
+    while (*p != '\0') {
+        char c = *(p++);
+
+        if (escaped) {
+            current += c;
+            escaped = false;
+            continue;
+        }
+
+        if (quotes) {
+            if (c == '\"') {
+                quotes = false;
+                continue;
+            }
+            if (c == '\\') {
+                escaped = true;
+                continue;
+            }
+            current += c;
+            continue;
+        }
+
+        if (std::isspace(c))
+            continue;
+
+        if (c == '\"') {
+            quotes = true;
+            had_quotes = true;
+            continue;
+        }
+
+        if (c == ',' || c == ')') {
+            if (!(c == ')' && current.empty() && !had_quotes))
+                args.push_back(current);
+            current = "";
+            had_quotes = false;
+            if (c == ')') {
+                done = true;
+                break;
+            }
+            continue;
+        }
+
+        current += c;
+    }
+
+    if (!done)
+        throw model_exception("Expected ')'");
+    while (*p != '\0' && std::isspace(*p))
+        p++;
+    if (*(p++) != '\0')
+        throw model_exception("Extra characters at the end of the string.");
+
+#define CHECK_NUM_ARGS(expected)   \
+    if (args.size() != (expected)) \
+        throw model_exception("Expected " + std::to_string(expected) + " arguments");
+#define CHECK_NUM_ARGS_RANGE(min, max)                                              \
+    if (args.size() < min || args.size() > max)                                     \
+        throw model_exception("Expected between " + std::to_string(min) + " and " + \
+          std::to_string(max) + " arguments");
+
+    /* Now actually parse the operation. */
+    /* FIXME: We currently only support unsigned numbers for keys and values. */
+    if (name == "begin_transaction") {
+        CHECK_NUM_ARGS(1);
+        return begin_transaction(parse_uint64(args[0]));
+    }
+    if (name == "checkpoint") {
+        CHECK_NUM_ARGS_RANGE(0, 1);
+        return checkpoint(args.size() == 0 ? nullptr : args[0].c_str());
+    }
+    if (name == "commit_transaction") {
+        CHECK_NUM_ARGS_RANGE(1, 3);
+        return commit_transaction(parse_uint64(args[0]),
+          args.size() < 1 ? k_timestamp_none : parse_uint64(args[1]),
```

Comment:
Line 140 already guarantees 1 <= args.size() <= 3.

Why are you testing for args.size() < 1 here? Either remove that check or change line 140 to CHECK_NUM_ARGS_RANGE(0, 3).

Reply:
How to remove the approval? I forgot one comment, this one, is a bug.

Reply:
Fixed in [70241f1](https://github.com/wiredtiger/wiredtiger/pull/10267/commits/70241f1a1fcf8c5b60f1e9303b46e4568d643a27). Thanks for catching it! I also added a test for this.

Reply:
And I will also undo your approval, so please re-review :) Thanks!